libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/CiMitUtilityService.java

```diff
@@ -68,8 +71,30 @@ private MitigationRoute getMitigationRoute(
                                         "No mitigation journey route event found."));
     }
 
+    public Optional<JourneyResponse> getMitigatedCiJourneyStep(ContraIndicator ci)
+            throws ConfigException, MitigationRouteConfigNotFoundException {
+        var cimitConfig = configService.getCimitConfig();
+        if (isCiMitigatable(ci)) {
+            return Optional.of(
+                    new JourneyResponse(
+                            getMitigationRoute(cimitConfig.get(ci.getCode()), ci.getDocument())
+                                    .event()));
+        }
+        return Optional.empty();
+    }
+
     private boolean isCiMitigatable(ContraIndicator ci) throws ConfigException {
         var cimitConfig = configService.getCimitConfig();
         return cimitConfig.containsKey(ci.getCode()) && !ci.isMitigated();
     }
+
+    public Optional<ContraIndicator> hasMitigatedContraIndicator(
+            ContraIndicators contraIndicators) {
+        for (var ci : contraIndicators.getContraIndicatorsMap().values()) {
+            if (ci.isMitigated()) {
+                return Optional.of(ci);
+            }
+        }
+        return Optional.empty();
+    }
```

Comment:
This might be a good place to use a stream
```suggestion
        return contraIndicators.getContraIndicatorsMap().values().stream()
                .filter(ContraIndicator::isMitigated)
                .findFirst();
    }
```